packages/flutter_test/lib/src/window.dart

```diff
@@ -944,3 +982,263 @@ class TestPlatformDispatcher implements ui.PlatformDispatcher {
+///     functionality.
+class TestFlutterView implements ui.FlutterView {
+  /// Constructs a [TestFlutterView] that defers all behavior to the given
+  /// [ui.FlutterView] unless explicitly overridden for testing.
+  TestFlutterView({
+    required ui.FlutterView view,
+    required TestPlatformDispatcher platformDispatcher,
+  }) : _view = view, _platformDispatcher = platformDispatcher {
+    _viewConfiguration = view.viewConfiguration.copyWith(view: this);
+  }
+
+  /// The [ui.FlutterView] backing this [TestFlutterView].
+  final ui.FlutterView _view;
+
+  final TestPlatformDispatcher _platformDispatcher;
+  late ui.ViewConfiguration _viewConfiguration;
+
+  @Deprecated(
+    'Use _viewConfiguration instead. '
+    'This is only available to enable backwards compatibility of TestWindow. '
+    'This feature was deprecated after v3.8.0.10.1.pre.'
+  )
+  ui.ViewConfiguration? _overrideConfiguration;
+
+  @override
+  TestPlatformDispatcher get platformDispatcher => _platformDispatcher;
+
+  @override
+  ui.ViewConfiguration get viewConfiguration => _overrideConfiguration ?? _viewConfiguration;
+
+  @Deprecated(
+    'Use setters for individual test properties instead. '
+    'This is only available to enable backwards compatibility of TestWindow. '
+    'This feature was deprecated after v3.8.0.10.1.pre.'
+  )
+  /// Sets the ViewConfiguration directly. Not supported publicly by
+  /// [TestFlutterView] to avoid ambiguity between setting the configuration
+  /// directly and using the test setters. Required temporarily for backwards
+  /// compatibility of [TestWindow].
+  // ignore: avoid_setters_without_getters
+  set _viewConfigurationLegacy(ui.ViewConfiguration? viewConfiguration) {
+    if (viewConfiguration == null) {
+      _overrideConfiguration = null;
+    } else {
+      _overrideConfiguration = viewConfiguration.copyWith(view: this);
+    }
+    platformDispatcher.onMetricsChanged?.call();
+  }
+
+  @override
+  Object get viewId => _view.viewId;
+
+  @override
+  /// The device pixel ratio to use for this test.
+  ///
+  /// Defaults to the value provided by [ui.FlutterView.devicePixelRatio]. This
+  /// can only be set in a test environment to emulate different view
+  /// configurations. A standard [ui.FlutterView] is not mutable from the framework.
+  ///
+  /// See also:
+  ///
+  ///   * [ui.FlutterView.devicePixelRatio] for the standard implementation
+  ///   * [resetDevicePixelRatio] to reset this value specifically
+  ///   * [reset] to reset all test values for this view
+  double get devicePixelRatio => _viewConfiguration.devicePixelRatio;
+  set devicePixelRatio(double devicePixelRatio) {
+    _viewConfiguration = _viewConfiguration.copyWith(devicePixelRatio: devicePixelRatio);
+    platformDispatcher.onMetricsChanged?.call();
+  }
+
+  /// Resets [devicePixelRatio] for this test view to the default value for this view.
+  void resetDevicePixelRatio() => devicePixelRatio = _view.devicePixelRatio;
```

Comment:
nit (here and elsewhere): our style is to not use `=>` for void methods since `=>` kinda implies a return.

Reply:
Think I got them all.